### ISSUE TITLE
Process is slow

### ISSUE THREAD

**Issue by mattdsteele** [mattdsteele/town-tester#2]
Look into ways of speeding up the process. A few choke points:
- Github API - Potentially cache the user search / repository results into a file so statistics can be generated manually?
- Repo checkout - Would a `svn export` be faster? We just need `HEAD`, not the entire repo.

**Comment by jerodsanto** [comment]
It will still be slow, but doing a shallow clone will make cloning faster, especially on repositories that have a lot of history. For instance:

```
git clone -q https://github.com/rails/rails
```

Takes 2m16s on my machine, while:

```
git clone -q --depth=1 https://github.com/rails/rails
```

Takes roughly 20s.

You can also get a list of remote refs without cloning with `git-ls-remote`, but I don't know if/how you can turn that into a list of filenames.

**Comment by mattdsteele** [comment]
@jerodsanto Makes sense; I didn't know about shallow clones. I've updated the script. Thanks!

Searching StackOverflow, it doesn't look like `git-ls-remote` will do the trick. I'll leave this issue open b/c I still think there are ways of making this faster; I just don't know what they are.
